Source/WebCore/page/Navigation.cpp

```diff
@@ -859,14 +866,17 @@ bool Navigation::innerDispatchNavigateEvent(NavigationNavigationType navigationT
             m_ongoingNavigateEvent->finish();
             m_ongoingNavigateEvent = nullptr;
 
-            // FIXME: Fill in error information.
+            ErrorInformation errorInformation;
             String errorMessage;
-            if (auto* error = jsDynamicCast<JSC::ErrorInstance*>(result))
-                errorMessage = makeString("Uncaught "_s, error->sanitizedToString(protectedScriptExecutionContext()->globalObject()));
+            if (auto* errorInstance = jsDynamicCast<JSC::ErrorInstance*>(result)) {
+                if (auto result = extractErrorInformationFromErrorInstance(protectedScriptExecutionContext()->globalObject(), *errorInstance)) {
+                    errorInformation = *result;
+                    errorMessage = makeString("Uncaught "_s, errorInformation.errorTypeString, ": "_s, errorInformation.message);
+                }
+            }
             auto exception = Exception(ExceptionCode::UnknownError, errorMessage);
 
-            // FIXME: Set line/column numbers.
-            dispatchEvent(ErrorEvent::create(eventNames().navigateerrorEvent, exception.message(), document->url().stringWithoutFragmentIdentifier(), 1, 1, { protectedScriptExecutionContext()->globalObject()->vm(), result }));
+            dispatchEvent(ErrorEvent::create(eventNames().navigateerrorEvent, errorMessage, errorInformation.sourceURL, errorInformation.line, errorInformation.column, { protectedScriptExecutionContext()->globalObject()->vm(), result }));
```

Comment:
Do we want to do this if there is no error information?

Reply:
Ditto.